src/clib-configure.c

```diff
@@ -181,6 +180,9 @@ int configure_package_with_manifest_name(const char *dir, const char *file) {
   pthread_mutex_unlock(&mutex);
 #endif
 
+  // Free the json if it was allocated before attempting to modify it
+  if (json != NULL) free(json);
```

Comment:
No need for this `if`, as `free(NULL)` works just fine 😉 

```suggestion
  free(json);
```